Experiment Vale Updates (#281)
Amplitude-Developer-Docs/amplitude-dev-center: docs/experiment/general/data-model.md

## Projects

Like Amplitude Analytics, Experiment uses projects. As a best practice, create a project per product and per environment. Because [flags](#flags-and-experiments), [experiments](#flags-and-experiments), and [deployments](#deployments) only exist within a single project, duplicate these objects across projects for the same product.

!!!tip "Copying a flag to another project"
    When developing a new feature with an experiment, you can create the experiment in the dev environment project to develop and test that the implementation is correct, then copy the experiment into the prod project to run the experiment in prod.

## Deployments

In Amplitude Experiment, a deployment serves a group of flags or experiments for use in an application. Deployments have an associated randomly generated **deployment key** (also known as an API key) which is used to identify the deployment and authorize requests to Amplitude Experiment's evaluation servers. 

!!!info "Client vs Server Deployments"

>     Deployments are either client or server deployments. Only server deployments can use access flag configs for [local evaluation](./evaluation/local-evaluation.md), and shouldn't be shared to made public in any way.

Author: @bgiori : >Only server deployments can use access flag configs for [local evaluation](./evaluation/local-evaluation.md), and shouldn't be shared to made public in any way.<

What's not meant to be shared or made public here? The deployment key or 'access flag configs' or just 'flag configs'?

Reviewer: Just "flag configs", original sentence had a typo

>     Deployments are either client or server deployments. Only server deployments can use access flag configs for [local evaluation](./evaluation/local-evaluation.md), and shouldn't be shared to made public in any way.

Reviewer: ```suggestion
    Deployments are either client or server deployments. Only server deployments can access flag configs for [local evaluation](./evaluation/local-evaluation.md), and shouldn't be shared to made public in any way.
```

## Deployments

In Amplitude Experiment, a deployment serves a group of flags or experiments for use in an application. Deployments have an associated randomly generated **deployment key** (also known as an API key) which Experiment uses to identify the deployment and authorize requests to the evaluation servers. 

!!!info "Client vs Server Deployments"
    Deployments are either client or server deployments. Only server deployments can use access flag configs for [local evaluation](./evaluation/local-evaluation.md), and shouldn't be shared to made public in any way.

Deployments live within Amplitude Analytics projects, and a project can have multiple deployments. 
Add deployments to [Flags and Experiments](../guides/getting-started/create-a-flag.md#add-a-deployment) in the same project. When Experiment's evaluation servers receive a request to fetch variants for a user, Experiment uses the deployment key to look up all associated flags and experiments for evaluation.

## Flags and experiments

> Experiment uses feature flag and experiments to serve a variable experience to a user. They're identified by the **flag key**, associated with `0-n` [deployments](#deployments), and contains `1-k` [variants](#variants). The **evaluation mode** (local or remote) determines whether the flag or experiment can be [locally evaluated](evaluation/local-evaluation.md) and may limit the targeting capabilities for the flag if set to local (local evaluation mode flags can't use advanced targeting features like behavioral cohorts).

Author: @bgiori : "They're identified by the **flag key**, associated with `0-n` [deployments](#deployments), and contains `1-k` [variants](#variants)." 

What is "they're" referring to in this sentence: feature flags/experiments, users, or variable experiences?

Reviewer: Nice catch, in this case it refers to "Feature flags and experiments".

## Projects

> Experiment uses the same projects which are required for Amplitude Analytics. Generally speaking you want to create a project per product and per environment. Since [flags](#flags-and-experiments), [experiments](#flags-and-experiments), and [deployments](#deployments) only exist within a single project, you will want to duplicate these objects across projects for the same product.

Reviewer: "Experiment uses the same projects which are required for Amplitude Analytics."

I still think this is a valuable piece of information.

## Deployments

In Amplitude Experiment, a deployment serves a group of flags or experiments for use in an application. Deployments have an associated randomly generated **deployment key** (also known as an API key) which Experiment uses to identify the deployment and authorize requests to the evaluation servers. 

!!!info "Client vs Server Deployments"
    Deployments are either client or server deployments. Only server deployments can use access flag configs for [local evaluation](./evaluation/local-evaluation.md), and shouldn't be shared to made public in any way.

Deployments live within Amplitude Analytics projects, and a project can have multiple deployments. 
Add deployments to [Flags and Experiments](../guides/getting-started/create-a-flag.md#add-a-deployment) in the same project. When Experiment's evaluation servers receive a request to fetch variants for a user, Experiment uses the deployment key to look up all associated flags and experiments for evaluation.

## Flags and experiments

> Experiment uses feature flag and experiments to serve a variable experience to a user. They're identified by the **flag key**, associated with `0-n` [deployments](#deployments), and contains `1-k` [variants](#variants). The **evaluation mode** (local or remote) determines whether the flag or experiment can be [locally evaluated](evaluation/local-evaluation.md) and may limit the targeting capabilities for the flag if set to local. Local evaluation mode flags can't use advanced targeting features like behavioral cohorts.

Reviewer: ```suggestion
Experiment uses feature flags and experiments to serve a variable experience to a user. They're identified by the **flag key**, associated with `0-n` [deployments](#deployments), and contains `1-k` [variants](#variants). The **evaluation mode** (local or remote) determines whether the flag or experiment can be [locally evaluated](evaluation/local-evaluation.md) and may limit the targeting capabilities for the flag if set to local. Local evaluation mode flags can't use advanced targeting features like behavioral cohorts.
```

|<div class='med-column'> Property </div>| Requirement | Description |
| --- | --- | --- |
| `Value` | **Required** | A string which identifies the variant in the instrumentation. The value string is checked for equality when a variant is accessed from the SDK or [Evaluation REST API](../apis/evaluation-api.md). Format must be lowercase kebab-case or snake_case. |
| `Payload` | Optional | Dynamic JSON payload for sending arbitrary data down with the variant. For example, you could send down a hex code to change the color of a component in your application. |

> | `Name` | Optional | Name for the variant. This like `Value`, but doesn't have formatting limitations, and you can change it without breaking the instrumentation in your code base.|

Reviewer: ```suggestion
| `Name` | Optional | Name for the variant. This is like `Value`, but doesn't have formatting limitations, and you can change it without breaking the instrumentation in your code base.|
```

| `Description` | Optional | A more detailed description of the variant. You can use this to describe what the user experiences when viewing the variable experience in more detail. |

!!!info "SDK Usage"
    Only the `Value` and `Payload` are available when accessing a variant from an SDK or the [Evaluation REST API](../apis/evaluation-api.md).

## Users

> Experiment users map to a user within Amplitude Analytics. Alongside flag configurations, users are an input to [evaluation](evaluation/implementation.md). Flag and experiment targeted rules make use of user properties.

Reviewer: ```suggestion
Experiment users map to a user within Amplitude Analytics. Alongside flag configurations, users are an input to [evaluation](evaluation/implementation.md). Flag and experiment targeting rules can make use of user properties.
```

> Amplitude Experiment passes users to evaluation via `fetch` requests in [remote evaluation](evaluation/remote-evaluation.md), or directly to the `evaluate` function for [local evaluation](evaluation/local-evaluation.md).

Reviewer: ```suggestion
Pass users to evaluation via `fetch` requests for [remote evaluation](evaluation/remote-evaluation.md), or directly to the `evaluate` function for [local evaluation](evaluation/local-evaluation.md).
```

| <div class='med-big-column'> Property </div> | Type | Description |
| --- | --- | --- |
| `user_id` | `string` | The [User ID](https://help.amplitude.com/hc/en-us/articles/206404628-Step-2-Identifying-your-users#h_533ee533-f04d-49d3-873f-5836945f43a6) is the primary identifier for the user, generally when the user is logged into an account within your system. Experiment uses the User ID when resolving the Amplitude ID on enrichment before [remote evaluation](evaluation/remote-evaluation.md) where the Amplitude ID is the default bucketing key. |
| `device_id` | `string` | The Device ID is the secondary identifier for the user. This is usually randomly generated by an analytics SDK on the client side or set in a cookie on the server side. The Device ID is also used when resolving the Amplitude ID on enrichment before [remote evaluation](evaluation/remote-evaluation.md) where the Amplitude ID is the default bucketing key. |
| `user_properties` | `object` | Optional object of custom properties used when evaluating the user during local or remote evaluation. |

> | `groups` | `object` | :material-beta: Optional object that lists groups associated with this user. Format is an object where the key is the group type, and the value is an array of strings, the group value for the type (for example, `{"org name":["Amplitude"]}`)  |

Reviewer: ```suggestion
| `groups` | `object` | :material-beta: Optional object that lists groups associated with this user. Format is an object where the key is the group type, and the value is an array of group value strings (for example, `{"org name":["Amplitude"]}`)  |
```

typo: sentence doesnt really make sense

| `Description` | Optional | A more detailed description of the variant. Can be used to describe what the user experiences when viewing the variable experience in more detail. |

!!!info "SDK Usage"
    Only the `Value` and `Payload` are available when accessing a variant from an SDK or the [Evaluation REST API](../apis/evaluation-api.md).

## Users

Experiment users map to a user within Amplitude Analytics. Alongside flag configurations, users are used as input to [evaluation](evaluation/implementation.md). User properties are used in flag and experiment targeting rules.

Within Amplitude Experiment, users are passed to evaluation via `fetch` requests in [remote evaluation](evaluation/remote-evaluation.md), or directly to the `evaluate` function for [local evaluation](evaluation/local-evaluation.md).

!!!warning
    **Either a user ID or device ID must be included in the user object for evaluation to succeed.** For example, remote evaluation returns a 400 error if both the User ID and Device ID are null, empty, or missing.

| <div class='med-big-column'> Property </div> | Type | Description |
| --- | --- | --- |

> | `user_id` | `string` | The [User ID](https://help.amplitude.com/hc/en-us/articles/206404628-Step-2-Identifying-your-users#h_533ee533-f04d-49d3-873f-5836945f43a6) is the primary identifier for the user, generally when the user is logged into an account within your system. The User ID is used when resolving the Amplitude ID on enrichment before [remote evaluation](evaluation/remote-evaluation.md) where the Amplitude ID is the default bucketing key. |

Author: @bgiori : "The User ID is the primary identifier for the user, generally when the user is logged into an account within your system:.

I'm finding this a little unclear. Does it mean:
"The User ID is the primary identifier for the user. This value typically comes from their user ID from your system" 

or does it mean something else?

Reviewer: Yeah, thats more clear. "The User ID is the primary identifier for the user. This value is typically their user ID within your system."
